bot/cogs/antimalware.py

```diff
@@ -17,37 +18,32 @@ def __init__(self, bot: Bot):
     @Cog.listener()
     async def on_message(self, message: Message) -> None:
         """Identify messages with prohibited attachments."""
-        rejected_attachments = False
-        detected_pyfile = False
+        if len(message.attachments) == 0:
+            return
+
+        embed = Embed()
         for attachment in message.attachments:
             if attachment.filename.lower().endswith('.py'):
-                detected_pyfile = True
+                embed.description = (
+                    "It looks like you tried to attach a Python file - please "
+                    f"use a code-pasting service such as [{PASTE_URL}]"
+                    f"({PASTE_URL}) instead."
+                )
                 break  # Other detections irrelevant because we prioritize the .py message.
             if not attachment.filename.lower().endswith(tuple(AntiMalwareConfig.whitelist)):
-                rejected_attachments = True
-
-        if detected_pyfile or rejected_attachments:
-            # Send a message to the user indicating the problem (with special treatment for .py)
-            author = message.author
-            if detected_pyfile:
-                msg = (
-                    f"{author.mention}, it looks like you tried to attach a Python file - please "
-                    f"use a code-pasting service such as https://paste.pythondiscord.com/ instead."
-                )
-            else:
                 meta_channel = self.bot.get_channel(Channels.meta)
-                msg = (
-                    f"{author.mention}, it looks like you tried to attach a file type we don't "
-                    f"allow. Feel free to ask in {meta_channel.mention} if you think this is a mistake."
+                embed.description = (
+                    "It looks like you tried to attach a file type that we "
+                    "do not allow. We currently allow the following file "
+                    f"types: **{', '.join(AntiMalwareConfig.whitelist)}**. \n\n"
+                    f"Feel free to ask in {meta_channel.mention} if you think "
+                    "this is a mistake."
                 )
-
-            await message.channel.send(msg)
+        if embed.description:
+            await message.channel.send(f"Hey {message.author.mention}!", embed=embed)
 
             # Delete the offending message:
-            try:
-                await message.delete()
-            except NotFound:
-                log.info(f"Tried to delete message `{message.id}`, but message could not be found.")
+            await message.delete()
```

Comment:
I notice that you removed the try/except, any special? Logging is useful

Reply:
I removed the try/except block because I don't think it's possible for it to error when the message is from the `on_message` event- Am I wrong in thinking so? I will happily add it back if I am 😁

Reply:
Yes, it is possible, as this is not the only `on_message` event with the potential to delete the new message.